### PR TITLE
Remove $tax_name property

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove property deprecated since PrestaShop 1.4
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| BC Break       | Product::$tax_name is removed.

### CODE REVIEW

**Review by kpodemski — APPROVED** [review]
It is a BC Break, but I find it similar to another that you have found where you removed a code that does nothing. This is why I accept this, but I'd like to see a review from another maintainer to validate this PR.

**Comment by matks** [comment]
> It is a BC Break, but I find it similar to another that you have found where you removed a code that does nothing. This is why I accept this, but I'd like to see a review from another maintainer to validate this PR.

The other PR removed a code that simply returned false. This PR removes a variable that might be used by modules, there's a difference 🤔

**Review by mflasquin — APPROVED** [review]
@matks, @kpodemski I think we can approve this even if there is a BC break because he target it on develop ?

**Pull request closed without merging**
